libs/jsonschema/instance.go

```diff
@@ -129,3 +133,34 @@ func (s *Schema) validatePattern(instance map[string]any) error {
 	}
 	return nil
 }
+
+func (s *Schema) validateConst(instance map[string]any) error {
+	for k, v := range instance {
+		fieldInfo, ok := s.Properties[k]
+		if !ok || fieldInfo.Const == nil {
+			continue
+		}
+		if v != fieldInfo.Const {
+			return fmt.Errorf("expected value of property %s to be %v. Found: %v", k, fieldInfo.Const, v)
+		}
+	}
+	return nil
+}
+
+// Validates that the instance matches at least one of the schemas in anyOf
+// but will also succeed if the property values are omitted.
+// For more information, see https://json-schema.org/understanding-json-schema/reference/combining#anyof.
+func (s *Schema) validateAnyOf(instance map[string]any) error {
+	if s.AnyOf == nil {
+		return nil
+	}
+	// Currently, we only validate const for anyOf schemas since anyOf is
+	// only used by skip_prompt_if, which only supports const.
+	for _, anyOf := range s.AnyOf {
+		err := anyOf.validateConst(instance)
```

Comment:
The `validateConst` function expects `s.Properties` to be non-nil. This is not guaranteed (I think), looking at the JSON schema spec. I understand we're only interested in validating the full config object (i.e. a string/string map) for the skip-prompt-if functionality here, but this seems brittle.

Having the `anyOf` property in the schema invites folks to use it as the JSON spec intends (i.e. also at the single field level) and that would cause panics here.

Reply:
I don't think it'll panic right? Maybe I am missing what you are pointing to, but a single field `anyOf` will simply be ignored in the current implementation.

Reply:
In the following (valid) example, the `properties` field is not set, and the code will panic.

```json
{
  "anyOf": [
    {
      "type": "string"
    }
  ]
}
```

Reply:
Tried it out, this does not panic. Accessing empty maps does not panic. You can try this out in the go playground:
```
	type apple struct {
		foo string
		bar map[string]any
	}

	fruit := apple{}
	v, ok := fruit.bar["key"]
	fmt.Printf("v: %v, ok: %v\n", v, ok)
```
This does not panic.

Reply:
Thanks, TIL!

Looked into the spec and it says:
> A nil map is equivalent to an empty map except that no elements may be added.

Reply:
Nice! Did not know this :)